### DISCUSSION TITLE
Welcome to Degoo Discussions!

### DISCUSSION THREAD

**Discussion by bernd-wechner** [bernd-wechner/Degoo#30] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nunyafb** [comment]
Hi Bernd, If I may use casual greeting. I'm Jim, I think it's obvious that Degoo was able to make very bod claimes about the amount of storage you would have available to you because they crippeled the ability to upload data. I wish I coud help you code but I don't have the skill set. If you need any other help (I don't know what) send me a message and I'll try to help. 
One thought I had crude that it is was to find for lack of a better word a stupid webbrowser set it up open it auto log into my account and Drag N' Drop the files. better yet something I could run on my nas that did the same thing. In any case I'll need to find the parts and tweak things together. I will make you a promise I'll send you a paypal donation once you get time to work on this and if you need someone to test give e a shout.

Jim P.

.

**Reply by iliabvf** [reply]
Hi, I found a solution, Im using Android x86 Virtual machine with FTP and backup soft:
file -> add extension to file .pdf -> backup upload to FTP -> FTP AndroidX86 Virtual machine -> automated upload to DEGOO

**Reply by nunyafb** [reply]
Again sorry for not responding to your suggestion, I simply didn't get notified (More likely I just missed it) someone had responded. May I rephrase your suggestion, 1) ADD ".PDF" to any file I want backed up. 2) Create an Android VM 3) obtain ANY/A specific backup application 4) have that application automatically backup all the PDF files. 

I hope I'm not not coming across snarky that's not my intention, I'm just trying to confirm your process. I look forward to hearing back from you.

Jim

**Comment by bernd-wechner** [comment]
@nunyafb Nice to hear from you. I admit I've stalled on this for lack of time,m one of my main next priorities being to explore resumeable uploads and downloads for large files interrupted by connectivity issues. That was the one feature I want before I can run this script as a nightly backup, with the caveat that I was also waiting on Degoo to hopefully provide me with some more info on ways to get a hash of the remote file. The problem is for nightly backups of my system I need the script to check if the local file is already backed up. Without a hash the best strategy is to check the last modification time of the local file and compare against the last modification time of the remote (uploaded file). That I have implemented and could roll out with, but it's not as robust as comparing a hash of the local and remote for identity.  Alas of course Degoo customer support is not all that responsive.

Drag'n'drop is a whole other level of complexity as  I have a simple CLI Python script that I'm hoping to use for nightly backups (schedule to run once nightly). For Drg'n'drop it would need to be a windowed program. This can be done, and added to this script, a simple GTK or QT window perhaps as a drag/drop target could be set up easily enough. But alas the second I or anyone creates such a window it begs more ... as in you'd want to add a tree view of the remote filesystem so you can select a folder to drop into for example and you'd want to add download and upload commands as well on menu, shortcut-menu, etc ... in short it's a bigger job.

My personal priority when If ind time would be to try and get uploads and downloads resumeable on interruption. That is a very common thing and the backend Degoo use is Google Cloud which does support that of course but they have a secret API key that we don't have and so what we can directly with the Google cloud backend is rather limited.  That's by design BTW, as they don't promise a Google cloud backend and have migrated once I believe, if I remember right early on the used Amazon storage.

Anyhow as it stands I'm aiming to hat resumable and then running under cron or a systemd schedule on a nightly basis some time. But it's slow as I took on a few too many side jobs and suddenly lost time to keep moving on this one. It's still on the radar, but not any time real soon alas ...

@iliabvf, Great solution. Alas I find the android app does not even reliably upload but hey .. if it's working for you, consider maybe a more beginner friendly how-to, posted here or as a gist linked to here etc. I'm sure some would be grateful.

**Reply by elsabio** [reply]
I'm just experimenting with automating Degoo for backup.  There is a 'comment' facility that seems to be attached to each uploaded file.  In your digging through protocol traces, do you have an understanding of how that interface works?  It seems to me that a backup client could compute the hash locally (either complete, or in blocks for performance) and upload it with the file as a comment, and in that way it could decide if an artefact had been changed and hence only upload when required.

How does the performance look for your current CLI model?  Would it support a FUSE mountpoint for example?  I have not had time to experiment with that yet.

**Reply by nunyafb** [reply]
I'm very sorry I haven't responded to you sooner. Please forgive me I'm not that type of person. I look forward to any practical solution to upload files, my suggestion was nothing more than WAG (Wild Ass Guess) as a possible way to move files. As I mentioned I don't have programing skills and have _**"Always relied on the kindness of strangers"**_ feel free to contact me directly degoo.discussions@yimsgu.com My email server will flag it and will see it for sure. Regards Jim

**Reply by bernd-wechner** [reply]
> I'm just experimenting with automating Degoo for backup. There is a 'comment' facility that seems to be attached to each uploaded file. In your digging through protocol traces, do you have an understanding of how that interface works? It seems to me that a backup client could compute the hash locally (either complete, or in blocks for performance) and upload it with the file as a comment, and in that way it could decide if an artefact had been changed and hence only upload when required.
> 
> How does the performance look for your current CLI model? Would it support a FUSE mountpoint for example? I have not had time to experiment with that yet.

Using the comment sounds like a great plan! Haven't seen it, haven't looked at this project in a while. But when I do, I'll check that out,

A FUSE implementation would rock. It turns out @MDKPredator has implemented one: https://github.com/MDKPredator/degoo_drive
I have yet to find time to test and play with t and perhaps merge back in here.  Also @buzzy was implementing (has implemented) a FUSE client in Go that also fixes the login issues.

Alas, the login issues are well understood and not easy to solve in Python yet: https://github.com/bernd-wechner/Degoo/issues/42 (it's due to a modern security feature Degoo enable to block any non-standard browsers - poo on them for that, called [JA3 fingerprinting](https://developers.cloudflare.com/bots/concepts/ja3-fingerprint), and Python cannot spoof a JA3 fingerprint yet alas though there are some work around).

**Comment by eric1194** [comment]
is it possible that Degoo change something, I don't seem to be able to login with the CLIs tool. I keep getting "login failed with: 403: Forbidden".

**Reply by bernd-wechner** [reply]
> is it possible that Degoo change something, I don't seem to be able to login with the CLIs tool. I keep getting "login failed with: 403: Forbidden".

A well understood problem with no immediate solution, alas. Discussed at length here: https://github.com/bernd-wechner/Degoo/issues/42
with some possible solutions to experiment with when time permits. Alas, the ideal solution is for the [Python SSL module](https://docs.python.org/3/library/ssl.html) and that may take some patience as it evolves.

**Comment by gmulligan** [comment]
has anyone made any progress.  it seems that curl-impersonate does also have a replacement lib-curl.  could we use that?

I did try using curl-impersonate-ff from the command line and it does seem to allow logins!!!

**Reply by buzzy** [reply]
I have been using my own fuse degoo filesystem for several months. It's working great. However, Degoo stopped allowing web-access for non-paying accounts so very few people have any use for these kind of tools now. Also, people can just use AirExplorer with Degoo, so not much point left in a CLI/fuse tool.

**Reply by bernd-wechner** [reply]
@buzzy, long time no hear, awesome to hear back from you. I was hoping you'd share your Golang solution some time. I can't see it on your list of repos. 

AirExplorer looks interesting but with the general sad lack of support Linux.

**Reply by bernd-wechner** [reply]
> I did try using curl-impersonate-ff from the command line and it does seem to allow logins!!!

I haven't had much time for this of late. But I'm guessing you mean this:

https://github.com/lwthiker/curl-impersonate/blob/main/INSTALL.md

(always post a link ;-). I'd like to give it a go some time. But time is tight in coming months again (still).

**Comment by Kreash** [comment]
After using Deggo for several years to backup photos, I encountered a problem downloading files. After communicating with technical support, they suggested that I download my entire database one file at a time manually.
After some indignation, I had the idea to write a script for uploading my files and learn a little graphQL.
It was only after the project was completed that I had the idea to look for similar projects.
Thanks for your project.
If someone finds it more convenient to use a browser than console CLI Util(s), then I’ll leave a link to my project https://github.com/Kreash/degoo-downloader
